File: 137-single-number-ii/137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        return rahulSir(nums);
    }
private:
    int rahulSir(vector<int>& nums) {
        int b1 = 0, b2 = 0;
        
        for(int n: nums) {
            b1 = (~b2) & (b1 ^ n);
            b2 = (~b1) & (b2 ^ n);
        }
        return b1;
    }
};

File: 137-single-number-ii/137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # Rahul sir
        
        b1, b2 = 0, 0
        
        for n in nums:
            b1 = (~b2) & (b1 ^ n)
            b2 = (~b1) & (b2 ^ n)
        
        return b1